mlflow/models/utils.py

```diff
@@ -863,3 +866,54 @@ def get_model_version_from_model_uri(model_uri):
     (name, version) = get_model_name_and_version(client, model_uri)
     model_version = client.get_model_version(name, version)
     return model_version
+
+
+def _enforce_params_schema(params: Optional[Dict[str, Any]], schema: Optional[ParamSchema]):
+    if params in [None, {}] and schema is None:
+        return params
+    if not isinstance(params, dict):
+        raise MlflowException(
+            f"Parameters must be a dictionary. Got type '{type(params).__name__}'."
+        )
+    if not isinstance(schema, ParamSchema):
+        raise MlflowException(
+            "Parameters schema must be an instance of ParamSchema. "
+            f"Got type '{type(schema).__name__}'."
+        )
+    if any(not isinstance(k, str) for k in params.keys()):
+        _logger.warning(
+            "Keys in parameters should be of type `str`, but received non-string keys."
+            "Converting all keys to string..."
+        )
+        params = {str(k): v for k, v in params.items()}
+
+    allowed_keys = {param.name for param in schema.params}
+    ignored_keys = set(params) - allowed_keys
+    _logger.warning(
+        f"Invalid arguments {list(ignored_keys)} are ignored for inference. "
+        f"Supported arguments are: {allowed_keys}. "
+        "To enable them, please add corresponding schema in ModelSignature."
+    )
```

Comment:
Let me know if I'm missing something, but it seems like this is always printed even if `ignored_keys` is empty. Can we only print it if `ignore_keys` contains elements?

cc @harupy @BenWilson2